src/test/java/com/google/api/gax/grpc/UnaryCallableTest.java

```diff
@@ -458,25 +462,106 @@ public Integer extractPageSize(Integer payload) {
     }
   }
 
-  private class PagedListResponse extends PagedListResponseImpl<Integer, List<Integer>, Integer> {
-    public PagedListResponse(
+  private class ListIntegersPagedResponse implements PagedListResponse<Integer> {
+
+    private final PagedListResponseContext<Integer, List<Integer>, Integer> context;
+    private final ListIntegersPage page;
+
+    public ListIntegersPagedResponse(
         UnaryCallable<Integer, List<Integer>> callable,
         PagedListDescriptor<Integer, List<Integer>, Integer> pageDescriptor,
         Integer request,
-        CallContext context) {
-      super(callable, pageDescriptor, request, context);
+        CallContext callContext) {
+      this.context = new PagedListResponseContext<>(callable, pageDescriptor, request, callContext);
+      this.page = new ListIntegersPage(this.context);
+    }
```

Comment:
If a `PagedListResponse` implementation just ends up forwarding everything to the context and page, maybe the better pattern would be to make an `AbstractPageListResponse`. Then, you would have:

```
class ListIntegersPagedResponse extends AbstractPageListResponse<Integer> {}
```

Yeah, I know, I generally thing abstract classes are an abomination, but I think it makes sense in this context.